### PR TITLE
docs: fix block.Header.Time description

### LINKED ISSUE #3514: docs: Fix incorrect note about header.Time in ABCI data types being the proposer's local time
**Tendermint version** `v0.31.1` with the patch from #3118

**ABCI app**: [Ercoin](https://gitlab.com/Ercoin/ercoin)

**What happened**:

If `create_empty_blocks` is set to `false` and a block is created, then time of that block is not the current time, but it refers to the time at which the previous block was created.

**What you expected to happen**:

Block time should be current. Past time may make transactions invalid if their validity is time-dependent. If a transaction is created with a current timestamp, then a block in which that transaction is included must not contain a lower time value in block header.

**Have you tried the latest version**: yes

**How to reproduce it** (as minimally and precisely as possible):

1. Set `create_empty_blocks` to `false` and `create_empty_blocks_interval` to a reasonably high value.
2. Observe the time of the block header.

### PR DESCRIPTION
**Pull request by melekes**
It's not proposer local time anymore, but a weighted median

Fixes #3514

* [x] Updated all relevant documentation in docs
* [ ] ~~Updated all code comments where relevant~~
* [ ] ~~Wrote tests~~
* [x] Updated CHANGELOG_PENDING.md

### CODE REVIEW

File: docs/spec/abci/abci.md [hunk 1]
```diff
@@ -347,8 +347,10 @@ Commit are included in the header of the next block.
   - `Version (Version)`: Version of the blockchain and the application
   - `ChainID (string)`: ID of the blockchain
   - `Height (int64)`: Height of the block in the chain
-  - `Time (google.protobuf.Timestamp)`: Time of the block. It is the proposer's
-    local time when block was created.
+  - `Time (google.protobuf.Timestamp)`: Time of the previous block.
```
Comment (milosevic) [inline docs/spec/abci/abci.md]: Isn't this time of the block (not previous block)?

Comment (melekes) [reply docs/spec/abci/abci.md]: No, we're using LastCommit to calculate median time https://github.com/tendermint/tendermint/blob/f965a4db15796dfca4e504d93eaa83760ced4af2/state/state.go#L149

Comment (melekes) [reply docs/spec/abci/abci.md]: We should've renamed `Time` to `LastTime` or something. But now given we're switching to proposer's local time (basically reverting the changes), it does not worth it.

Comment (milosevic) [reply docs/spec/abci/abci.md]: We use LastCommit of block h to calculate time for block h+1. But it's still time of the current block (and shouldn't be renamed to LastTime).

**Comment by b00f** [comment]
It was good if the label also changed to `PreviousBlockTime`. A little bit confusing.

**Comment by melekes** [comment]
We're going to switch back to proposer-based timestamp https://github.com/tendermint/tendermint/issues/2840. Sorry for confusion

**Pull request merged**